capa/features/extractors/ts/extractor.py

```diff
@@ -0,0 +1,105 @@
+from typing import List, Tuple, Union, Iterator
+
+from tree_sitter import Node
+
+import capa.features.extractors.script
+import capa.features.extractors.ts.file
+import capa.features.extractors.ts.engine
+import capa.features.extractors.ts.global_
+import capa.features.extractors.ts.function
+from capa.features.common import Namespace
+from capa.features.address import NO_ADDRESS, Address, AbsoluteVirtualAddress, FileOffsetRangeAddress
+from capa.features.extractors.script import LANG_TEM, LANG_HTML
+from capa.features.extractors.ts.engine import TreeSitterHTMLEngine, TreeSitterTemplateEngine, TreeSitterExtractorEngine
+from capa.features.extractors.ts.function import PSEUDO_MAIN, TSFunctionInner
+from capa.features.extractors.base_extractor import Feature, BBHandle, InsnHandle, FunctionHandle, FeatureExtractor
+
+
+class TreeSitterFeatureExtractor(FeatureExtractor):
+    code_sections: List[TreeSitterExtractorEngine]
```

Comment:
Can `code_sections` be renamed `engines` to make it more intuitive for the reader?